gserver/cmtclient.go

```diff
@@ -0,0 +1,193 @@
+	cmtclient "github.com/cometbft/cometbft/rpc/client"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	cmttypes "github.com/cometbft/cometbft/types"
+	client "github.com/cosmos/cosmos-sdk/client"
+	ggrpc "github.com/gordian-engine/gcosmos/gserver/internal/ggrpc"
+	"github.com/spf13/cobra"
+	grpc "google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var _ client.CometRPC = (*Client)(nil)
+
+type Client struct {
+	cmd     *cobra.Command
+	gclient ggrpc.GordianGRPCClient
+}
+
+func NewClient(
+	cmd *cobra.Command,
+	grpcAddress string,
+	grpcInsecure bool,
+) (*Client, error) {
+	var dialOpts []grpc.DialOption
+	if grpcInsecure {
+		dialOpts = append(dialOpts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	}
+	cc, err := grpc.NewClient(grpcAddress, dialOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial gRPC: %w", err)
+	}
+
+	return &Client{
+		cmd:     cmd,
+		gclient: ggrpc.NewGordianGRPCClient(cc),
+	}, nil
+}
+
+func (c *Client) ABCIInfo(ctx context.Context) (*coretypes.ResultABCIInfo, error) {
+	panic(fmt.Errorf("not implemented"))
+}
+
+func (c *Client) ABCIQuery(ctx context.Context, path string, data bytes.HexBytes) (*coretypes.ResultABCIQuery, error) {
+	panic(fmt.Errorf("not implemented"))
+}
+
+func (c *Client) ABCIQueryWithOptions(ctx context.Context, path string, data bytes.HexBytes, opts cmtclient.ABCIQueryOptions) (*coretypes.ResultABCIQuery, error) {
+	panic(fmt.Errorf("not implemented"))
+}
+
+func (c *Client) BroadcastTxAsync(ctx context.Context, tx cmttypes.Tx) (*coretypes.ResultBroadcastTx, error) {
+	// TODO implement
+	return c.BroadcastTxSync(ctx, tx)
+}
+
+func (c *Client) BroadcastTxCommit(ctx context.Context, tx cmttypes.Tx) (*coretypes.ResultBroadcastTxCommit, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+func (c *Client) BroadcastTxSync(ctx context.Context, tx cmttypes.Tx) (*coretypes.ResultBroadcastTx, error) {
+	clientCtx, err := client.GetClientTxContext(c.cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get client tx context: %w", err)
+	}
+
+	sdkTx, err := clientCtx.TxConfig.TxDecoder()(tx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode tx: %w", err)
+	}
+
+	txJson, err := clientCtx.TxConfig.TxJSONEncoder()(sdkTx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode tx: %w", err)
+	}
+
+	res, err := c.gclient.SimulateTransaction(ctx, &ggrpc.SubmitSimulationTransactionRequest{
+		Tx: txJson,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to simulate transaction: %w", err)
+	}
+	if res.Error != "" {
+		return nil, fmt.Errorf("failed to simulate transaction: %s", res.Error)
+	}
+
+	res, err = c.gclient.SubmitTransaction(ctx, &ggrpc.SubmitTransactionRequest{
+		Tx: txJson,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to submit transaction: %w", err)
+	}
+	if res.Error != "" {
+		return nil, fmt.Errorf("failed to submit transaction: %s", res.Error)
```

Comment:
Use a different message from the above error, maybe "failure expecting transaction"?